### PR TITLE
RCAL-787: allow DQ flags to be specified by the user in resample.

### PR DESCRIPTION
**Pull request by mairanteodoro**
Resolves [RCAL-787](https://jira.stsci.edu/browse/RCAL-787)

This PR extends the resample spec parameters to allow for user-provided DQ flags to be used when creating the resampling mask.

The changes also include refactoring the unit tests to comply with the new split between L2 and L3 files.

**Regression test**
~https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/694/~
~https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/698/~
https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/699/

**Checklist**
- [X] added entry in `CHANGES.rst` under the corresponding subsection
- [X] updated relevant tests
- [X] updated relevant documentation
- [ ] updated relevant milestone(s)
- [ ] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below. [How to run regression tests on a PR](https://github.com/spacetelescope/romancal/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

**Review by tddesjardins — APPROVED** [review]
Looks good to me re: adding user input for good bits. I'll let others speak to the test changes. Thanks, Mairan!

File: romancal/resample/resample_utils.py [hunk 1]
```diff
@@ -170,7 +174,7 @@ def build_mask(dqarr, bitvalue):
     ----------
     dqarr : numpy.ndarray
         Input DQ array.
-    bitvalue : int
+    bitvalue : str
         Bitvalue flag.
```
Comment (schlafly) [inline romancal/resample/resample_utils.py:178]: Let's also change the doc to be a "bitvalue flag string" or something, and maybe 'see astropy.nddata.bitmask.interpret_bit_flags for details.'

**Comment by schlafly** [comment]
Can you also say a word or two about the resample_data = True vs. False changes?

**Comment by mairanteodoro** [comment]
> Can you also say a word or two about the resample_data = True vs. False changes?

The unit tests for which the change was applied do not require resampling.

**Comment by schlafly** [comment]
Maybe elaborating, you used to use resample in those unit tests, and now you don't?  We could have continued resampling those images but for ~speed purposes you have now disabled that?  For real data we ~always do want to resample, and there are some tests in there still that do resample the input data (?), but these now don't?  I'm just trying to figure out what the motivation here was and whether it's connected to either the units changes or the bit customization change or what.

**Comment by mairanteodoro** [comment]
> Maybe elaborating, you used to use resample in those unit tests, and now you don't? We could have continued resampling those images but for ~speed purposes you have now disabled that? For real data we ~always do want to resample, and there are some tests in there still that do resample the input data (?), but these now don't? I'm just trying to figure out what the motivation here was and whether it's connected to either the units changes or the bit customization change or what.

There was no other motivation other than the tests did not need resampling.

I reverted those changes and now we're back to what it was before (default behavior `resample_data=True`.)

**Comment by schlafly** [comment]
Thanks.  Okay, this looks good to me.  Let's get the tests passing; maybe just merge main back in and rerun?

File: CHANGES.rst [hunk 2]
```diff
@@ -39,6 +39,11 @@ resample
 --------
 
 - Update location of ``basic`` attributes. [#1131]
+- Allow user to provide DQ flags to use/exclude when creating resampling mask. [#1166]
+
+- Updated Level 3 ``cal_step`` attribute creation. [#1165]
+
+- Updated Level 3 ``cal_step`` attribute creation. [#1165]
```
Comment (schlafly) [inline CHANGES.rst:46]: Can you deduplicate this?

**Review by schlafly — APPROVED** [review]
Looks good to me.  Can you fix the change log and we'll merge?

**Pull request merged**